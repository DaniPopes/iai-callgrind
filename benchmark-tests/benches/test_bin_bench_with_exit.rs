use iai_callgrind::{main, ExitWith, Options};

main!(
    run = cmd = "benchmark-tests-exit",
    opts = Options::default().exit_with(ExitWith::Success),
    id = "succeed", args = ["0"];
    run = cmd = "benchmark-tests-exit",
          opts = Options::default().exit_with(ExitWith::Failure),
          id = "fail_with_1", args = ["1"],
          id = "fail_with_2", args = ["2"],
          id = "fail_with_3", args = ["3"],
          id = "fail_with_255", args = ["255"];
    run = cmd = "benchmark-tests-exit",
          opts = Options::default().exit_with(ExitWith::Code(0)),
          id = "code_0", args = ["0"];
    run = cmd = "benchmark-tests-exit",
          opts = Options::default().exit_with(ExitWith::Code(1)),
          id = "code_1", args = ["1"];
    run = cmd = "benchmark-tests-exit",
          opts = Options::default().exit_with(ExitWith::Code(2)),
          id = "code_2", args = ["2"];
    run = cmd = "benchmark-tests-exit",
          opts = Options::default().exit_with(ExitWith::Code(3)),
          id = "code_3", args = ["3"];
    run = cmd = "benchmark-tests-exit",
          opts = Options::default().exit_with(ExitWith::Code(255)),
          id = "code_255", args = ["-1"];
);
