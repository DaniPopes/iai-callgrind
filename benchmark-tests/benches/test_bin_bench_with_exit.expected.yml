runs:
  - data:
      - group: test_exits
        function: benchmark-tests-exit
        id: code_0
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_0.log
            - callgrind.benchmark-tests-exit.code_0.out
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: code_1
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_1.log
            - callgrind.benchmark-tests-exit.code_1.out
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: code_2
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_2.log
            - callgrind.benchmark-tests-exit.code_2.out
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: code_255
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_255.log
            - callgrind.benchmark-tests-exit.code_255.out
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: code_3
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_3.log
            - callgrind.benchmark-tests-exit.code_3.out
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: fail_with_1
        expected:
          files:
            - callgrind.benchmark-tests-exit.fail_with_1.log
            - callgrind.benchmark-tests-exit.fail_with_1.out
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: fail_with_2
        expected:
          files:
            - callgrind.benchmark-tests-exit.fail_with_2.log
            - callgrind.benchmark-tests-exit.fail_with_2.out
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: fail_with_255
        expected:
          files:
            - callgrind.benchmark-tests-exit.fail_with_255.log
            - callgrind.benchmark-tests-exit.fail_with_255.out
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: fail_with_3
        expected:
          files:
            - callgrind.benchmark-tests-exit.fail_with_3.log
            - callgrind.benchmark-tests-exit.fail_with_3.out
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: succeed
        expected:
          files:
            - callgrind.benchmark-tests-exit.succeed.log
            - callgrind.benchmark-tests-exit.succeed.out
            - summary.json
  - data:
      - group: test_exits
        function: benchmark-tests-exit
        id: code_0
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_0.log
            - callgrind.benchmark-tests-exit.code_0.log.old
            - callgrind.benchmark-tests-exit.code_0.out
            - callgrind.benchmark-tests-exit.code_0.out.old
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: code_1
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_1.log
            - callgrind.benchmark-tests-exit.code_1.log.old
            - callgrind.benchmark-tests-exit.code_1.out
            - callgrind.benchmark-tests-exit.code_1.out.old
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: code_2
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_2.log
            - callgrind.benchmark-tests-exit.code_2.log.old
            - callgrind.benchmark-tests-exit.code_2.out
            - callgrind.benchmark-tests-exit.code_2.out.old
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: code_255
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_255.log
            - callgrind.benchmark-tests-exit.code_255.log.old
            - callgrind.benchmark-tests-exit.code_255.out
            - callgrind.benchmark-tests-exit.code_255.out.old
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: code_3
        expected:
          files:
            - callgrind.benchmark-tests-exit.code_3.log
            - callgrind.benchmark-tests-exit.code_3.log.old
            - callgrind.benchmark-tests-exit.code_3.out
            - callgrind.benchmark-tests-exit.code_3.out.old
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: fail_with_1
        expected:
          files:
            - callgrind.benchmark-tests-exit.fail_with_1.log
            - callgrind.benchmark-tests-exit.fail_with_1.log.old
            - callgrind.benchmark-tests-exit.fail_with_1.out
            - callgrind.benchmark-tests-exit.fail_with_1.out.old
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: fail_with_2
        expected:
          files:
            - callgrind.benchmark-tests-exit.fail_with_2.log
            - callgrind.benchmark-tests-exit.fail_with_2.log.old
            - callgrind.benchmark-tests-exit.fail_with_2.out
            - callgrind.benchmark-tests-exit.fail_with_2.out.old
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: fail_with_255
        expected:
          files:
            - callgrind.benchmark-tests-exit.fail_with_255.log
            - callgrind.benchmark-tests-exit.fail_with_255.log.old
            - callgrind.benchmark-tests-exit.fail_with_255.out
            - callgrind.benchmark-tests-exit.fail_with_255.out.old
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: fail_with_3
        expected:
          files:
            - callgrind.benchmark-tests-exit.fail_with_3.log
            - callgrind.benchmark-tests-exit.fail_with_3.log.old
            - callgrind.benchmark-tests-exit.fail_with_3.out
            - callgrind.benchmark-tests-exit.fail_with_3.out.old
            - summary.json
      - group: test_exits
        function: benchmark-tests-exit
        id: succeed
        expected:
          files:
            - callgrind.benchmark-tests-exit.succeed.log
            - callgrind.benchmark-tests-exit.succeed.log.old
            - callgrind.benchmark-tests-exit.succeed.out
            - callgrind.benchmark-tests-exit.succeed.out.old
            - summary.json
